src/hooks/auditoriumSections.ts

```diff
@@ -0,0 +1,134 @@
+import { User } from "types/User";
+
+import { setGridData } from "api/profile";
+
+import {
+  currentAuditoriumSectionsSelector,
+  currentAuditoriumSectionsByIdSelector,
+} from "utils/selectors";
+import { WithId } from "utils/id";
+import { getPositionHash } from "utils/auditorium";
+
+import { useSelector } from "./useSelector";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
+import { useRecentVenueUsers } from "./users";
+import { useUser } from "./useUser";
+import { useCallback, useMemo } from "react";
+
+const useConnectAuditoriumSections = (venueId?: string) => {
+  useFirestoreConnect(() => {
+    if (!venueId) return [];
+
+    return [
+      {
+        collection: "venues",
+        doc: venueId,
+        subcollections: [{ collection: "sections" }],
+        storeAs: "currentAuditoriumSections",
+      },
+    ];
+  });
+};
+
+export interface UseAuditoriumSectionProps {
+  sectionId: string;
+  venueId?: string;
+}
+
+export const useAuditoriumSection = ({
+  venueId,
+  sectionId,
+}: UseAuditoriumSectionProps) => {
+  useConnectAuditoriumSections(venueId);
+
+  const { userWithId } = useUser();
+  const userId = userWithId?.id;
+
+  const sections = useSelector(currentAuditoriumSectionsByIdSelector);
+  const section = sections?.[sectionId];
+
+  const seatedUsers = useSectionSeatedUsers(venueId, sectionId);
+
+  const seatedUsersByHash = useMemo(
+    () =>
+      seatedUsers.reduce<Record<string, WithId<User> | undefined>>(
+        (acc, user) => {
+          if (!venueId) return acc;
+
+          const takenRow = user.data?.[venueId].row;
+          const takenColumn = user.data?.[venueId].column;
```

Comment:
Do we need the safe accessors 'deeper in the chain' since we're using them 'earlier'?

```suggestion
          const takenRow = user.data?.[venueId]?.row;
          const takenColumn = user.data?.[venueId]?.column;
```